null string exception on running run tpra region regenerate all

* Using build 2.2.1, after a fresh install of TP, when I run tpra region regenerate all, I get an error like this
```
Caused by: java.lang.NumberFormatException: Cannot parse null string
        at java.base/java.lang.Integer.parseInt(Integer.java:526) ~[?:?]
        at java.base/java.lang.Integer.parseInt(Integer.java:661) ~[?:?]
        at TownyProvinces-2.2.1.jar/io.github.townyadvanced.townyprovinces.settings.TownyProvincesSettings.getAverageProvinceSize(TownyProvincesSettings.java:101) 
```
* I investigated this, and it is casued by FileMgmt.loadFileIntoHashMap being unable to load a yml file. So none of the region definitions get loaded.
* It used to do this, but now I'm pretty sure it only accepts java "properties" format.
* In the very short term users can do a quick fix probably by converting their region definition .yml file formats to use "=" instead of ":"
* But we probably shouldn't consider this a fix, because we should probably support existing users file formats.
* This can be done simply by adding a yml parser type of thing

--- a/pom.xml
+++ b/pom.xml
@@ -11,7 +11,7 @@
     <java.version>1.16</java.version>
     <project.bukkitAPIVersion>1.15</project.bukkitAPIVersion>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <towny.version>0.101.2.5</towny.version>
+    <towny.version>0.102.0.0</towny.version>
   </properties>
 
   <repositories>
@@ -38,32 +38,18 @@
   </repositories>
 
   <dependencies>
-
-    <dependency>
-      <groupId>io.papermc.paper</groupId>
-      <artifactId>paper-api</artifactId>
-      <version>1.21.8-R0.1-SNAPSHOT</version>
-      <scope>provided</scope>
-    </dependency>
-<!--    <dependency>
-      <groupId>com.palmergames.bukkit.towny</groupId>
-      <artifactId>towny</artifactId>
-      <version>${towny.version}</version>
-      <scope>provided</scope>
-    </dependency>-->
-    <dependency>
-      <groupId>com.github.TownyAdvanced.Towny</groupId>
-      <artifactId>towny</artifactId>
-      <version>master-4907b3123a-1</version>
-	</dependency>
-      <!--
       <dependency>
-          <groupId>us.dynmap</groupId>
-          <artifactId>dynmap-api</artifactId>
-          <version>2.5</version>
+          <groupId>io.papermc.paper</groupId>
+          <artifactId>paper-api</artifactId>
+          <version>1.21.8-R0.1-SNAPSHOT</version>
           <scope>provided</scope>
       </dependency>
-      -->
+      <dependency>
+          <groupId>com.palmergames.bukkit.towny</groupId>
+          <artifactId>towny</artifactId>
+          <version>${towny.version}</version>
+          <scope>provided</scope>
+      </dependency>
       <dependency>
           <groupId>us.dynmap</groupId>
           <artifactId>dynmap-api</artifactId>

--- a/src/main/java/io/github/townyadvanced/townyprovinces/TownyProvinces.java
+++ b/src/main/java/io/github/townyadvanced/townyprovinces/TownyProvinces.java
@@ -42,7 +42,7 @@ public class TownyProvinces extends JavaPlugin {
 	public static final Object PRICE_RECALCULATION_JOB_LOCK = new Object();
 	public static final Object MAP_DISPLAY_JOB_LOCK = new Object();
 	private static TownyProvinces plugin;
-	private static final Version requiredTownyVersion = Version.fromString("0.101.2.5");
+	private static final Version requiredTownyVersion = Version.fromString("0.102.0.0");
 	private final TaskScheduler scheduler;
 
 	public TownyProvinces() {

--- a/src/main/java/io/github/townyadvanced/townyprovinces/jobs/map_display/MapDisplayTaskController.java
+++ b/src/main/java/io/github/townyadvanced/townyprovinces/jobs/map_display/MapDisplayTaskController.java
@@ -27,7 +27,8 @@ public class MapDisplayTaskController {
 	
 	public static void reloadIntegrations() {
 		
-		mapDisplayTask.cancel();
+		if (mapDisplayTask != null)
+			mapDisplayTask.cancel();
 		synchronized (TownyProvinces.MAP_DISPLAY_JOB_LOCK) {
 			synchronized (TownyProvinces.REGION_REGENERATION_JOB_LOCK) {
 				synchronized (TownyProvinces.PRICE_RECALCULATION_JOB_LOCK) {

--- a/src/main/java/io/github/townyadvanced/townyprovinces/objects/Region.java
+++ b/src/main/java/io/github/townyadvanced/townyprovinces/objects/Region.java
@@ -1,8 +1,9 @@
 package io.github.townyadvanced.townyprovinces.objects;
 
 import com.palmergames.bukkit.towny.object.Coord;
-import com.palmergames.util.FileMgmt;
 import io.github.townyadvanced.townyprovinces.settings.TownyProvincesSettings;
+import io.github.townyadvanced.townyprovinces.util.FileUtil;
+
 import org.bukkit.Location;
 
 import java.io.File;
@@ -36,7 +37,7 @@ public class Region {
 	private Set<Province> provinces; //Set of provinces. Assigned when required
 	
 	public Region(File regionDefinitionFile) {
-		Map<String, String> regionDefinitions = FileMgmt.loadFileIntoHashMap(regionDefinitionFile);
+		Map<String, String> regionDefinitions = FileUtil.loadFileIntoHashMap(regionDefinitionFile);
 		this.name = TownyProvincesSettings.getRegionName(regionDefinitions);
 		this.brushSquareRadiusInChunks = TownyProvincesSettings.getBrushSquareRadiusInChunks(regionDefinitions);
 		this.provinceSquareRadius = TownyProvincesSettings.calculateProvinceSquareRadius(regionDefinitions);

--- a/src/main/java/io/github/townyadvanced/townyprovinces/util/FileUtil.java
+++ b/src/main/java/io/github/townyadvanced/townyprovinces/util/FileUtil.java
@@ -1,18 +1,25 @@
 package io.github.townyadvanced.townyprovinces.util;
 
+import com.palmergames.bukkit.towny.Towny;
 import com.palmergames.util.FileMgmt;
 import io.github.townyadvanced.townyprovinces.TownyProvinces;
 import io.github.townyadvanced.townyprovinces.messaging.Messaging;
 import org.bukkit.Bukkit;
 
 import java.io.File;
+import java.io.FileInputStream;
 import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Properties;
+import java.util.logging.Level;
 
 public class FileUtil {
 
@@ -123,5 +130,23 @@ public class FileUtil {
 			e.printStackTrace();
 			return false;
 		}
+	}
+
+	public static HashMap<String, String> loadFileIntoHashMap(File file) {
+
+		HashMap<String, String> keys = new HashMap<>();
+		try (FileInputStream fis = new FileInputStream(file);
+			InputStreamReader isr = new InputStreamReader(fis, StandardCharsets.UTF_8)) {
+			Properties properties = new Properties();
+			properties.load(isr);
+			for (String key : properties.stringPropertyNames()) {
+				String value = properties.getProperty(key);
+				keys.put(key, String.valueOf(value));
+			}
+		} catch (IOException e) {
+			Towny.getPlugin().getLogger().log(Level.WARNING, "An exception occurred while reading file " + file.getName(), e);
+		}
+		return keys;
+
 	}
 }

